pkg/pool-weighted/contracts/managed/vendor/BaseWeightedPool.sol

```diff
@@ -75,31 +75,35 @@ abstract contract BaseWeightedPool is IMinimalSwapInfoPool, BasePool {
     /**
      * @dev Returns the normalized weight of `token`. Weights are fixed point numbers that sum to FixedPoint.ONE.
      */
-    function _getNormalizedWeight(IERC20 token) internal view virtual returns (uint256);
+    function _getNormalizedWeight(IERC20 token, uint256 weightChangeProgress) internal view virtual returns (uint256);
+
+    /**
+     * @notice Returns all normalized weights, in the same order as the Pool's tokens.
+     */
+    function getNormalizedWeights() public view returns (uint256[] memory) {
```

Comment:
```suggestion
    function getNormalizedWeights() external view returns (uint256[] memory) {
```

So we can be sure we never call this version ourselves

Reply:
We do actually call this ourselves in cases where we don't have the token array onhand. Prior to [6856979](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1769/commits/6856979b4a2d85d22cb84ed698e6578c03a3846e) we did have this external but it ballooned bytecode size.